src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -1486,22 +1486,28 @@ public int Compare(Symbol fst, Symbol snd)
                         var srcSymbol = symbols[best.Index];
                         var mdSymbol = symbols[secondBest.Index];
 
-                        object arg0;
-
-                        if (best.IsFromSourceModule)
-                        {
-                            arg0 = srcSymbol.Locations.First().SourceTree.FilePath;
-                        }
-                        else
-                        {
-                            Debug.Assert(best.IsFromAddedModule);
-                            arg0 = srcSymbol.ContainingModule;
-                        }
-
                         //if names match, arities match, and containing symbols match (recursively), ...
                         if (srcSymbol.ToDisplayString(SymbolDisplayFormat.QualifiedNameArityFormat) ==
```

Comment:
>srcSymbol.ToDisplayString(SymbolDisplayFormat.QualifiedNameArityFormat) [](start = 28, length = 71)

I think we should use less expensive check, the one that doesn't rely on ToDisplayString.

Reply:
> I think we should use less expensive check, the one that doesn't rely on ToDisplayString.

This looks like an existing code was simply moved.

---
In reply to: [215090529](https://github.com/dotnet/roslyn/pull/29180#discussion_r215090529) [](ancestors = 215090529)